maven/plugins/com.liferay.ide.maven.core/src/com/liferay/ide/maven/core/NewGradleJSFModuleProjectProvider.java

```diff
@@ -0,0 +1,224 @@
+
+            final String[] gav = archetypeArtifactId.split( ":" );
+
+            final String archetypeVersion = gav[gav.length - 1];
+
+            archetype.setGroupId( gav[0] );
+            archetype.setArtifactId( gav[1] );
+            archetype.setVersion( archetypeVersion );
+
+            Artifact mavenArchtype = AetherUtil.getMavenArchtype(
+                archetype.getGroupId(), archetype.getArtifactId(), archetype.getVersion() );
+
+            final ResolverConfiguration resolverConfig = new ResolverConfiguration();
+            ProjectImportConfiguration configuration = new ProjectImportConfiguration( resolverConfig );
+            final Properties properties = new Properties();
+
+            IPath projectLocation = createArchetypeProject(
+                mavenArchtype, location, archetype, groupId, artifactId, version, "com.example", properties,
+                configuration, monitor );
+
+            IPath buildGradlePath = projectLocation.append( "build.gradle" );
+
+            if( buildGradlePath.toFile().exists() )
+            {
+                try
+                {
+                    boolean hasLiferayWorkspace = LiferayWorkspaceUtil.hasLiferayWorkspace();
+
+                    if( hasLiferayWorkspace )
+                    {
+                        List<String> buildGradleContents =
+                            Files.readAllLines( Paths.get( buildGradlePath.toFile().toURI() ), StandardCharsets.UTF_8 );
+                        List<String> modifyContents = new ArrayList<String>();
+
+                        for( String line : buildGradleContents )
+                        {
+                            if( line.startsWith( "apply" ) )
+                            {
+                                continue;
+                            }
+
+                            modifyContents.add( line );
+                        }
+
+                        Files.write( buildGradlePath.toFile().toPath(), modifyContents, StandardCharsets.UTF_8 );
+                    }
+                }
+                catch( Exception e )
+                {
+                    ProjectCore.logError( "Failed to check LiferayWorkspace project. " );
+                }
+            }
+
+            IPath buildPom = projectLocation.append( "pom.xml" );
+
+            if( buildPom.toFile().exists() )
+            {
+                buildPom.toFile().delete();
+            }
+
+            ILiferayProjectImporter importer = LiferayCore.getImporter( "gradle" );
+            IStatus canImport = importer.canImport( projectLocation.toOSString() );
+
+            if( canImport.getCode() != Status.ERROR )
+            {
+                importer.importProjects( projectLocation.toOSString(), monitor );
+            }
+
+        }
+        catch( Exception e )
+        {
+            throw new CoreException( LiferayCore.createErrorStatus( e ) );
+        }
+
+        return retval;
+    }
+
+    private IPath createArchetypeProject(
```

Comment:
This code I extracted it from m2e code and modify its parmarter. This code passed the archtype we manually download from maven central, this function will be automatically done by m2e code. It just implemented creating project and didn't include maven import project section code. So its different from the standare m2e createArchetypeProjects function.